lms/djangoapps/badges/handlers.py

```diff
@@ -18,3 +21,46 @@ def award_badge_on_enrollment(sender, event=None, user=None, **kwargs):  # pylin
     """
     if badges_enabled and event == EnrollStatusChange.enroll:
         award_enrollment_badge(user)
+
+
+@receiver(post_save, sender=BadgeAssertion)
+def update_leaderboard_entry(sender, instance, **kwargs):
+    """
+    Update or create a leaderboard entry when a BadgeAssertion is saved.
+    """
+    user = instance.user
+    badges = BadgeAssertion.objects.filter(user=user)
+
+    course_badge_score, event_badge_score = LeaderboardConfiguration.get_current_or_default_values()
+
+    course_badge_count = badges.filter(badge_class__issuing_component='').count()
+    event_badge_count = badges.filter(badge_class__issuing_component='openedx__course').count()
+
+    leaderboard_entry, created = LeaderboardEntry.objects.get_or_create(user=user)
+    leaderboard_entry.badge_count = badges.count()
+    leaderboard_entry.event_badge_count = event_badge_count
+    leaderboard_entry.course_badge_count = course_badge_count
+
+    leaderboard_entry.score = calculate_score(
+        course_badge_score,
+        event_badge_score,
+        course_badge_count,
+        event_badge_count
+    )
+
+    leaderboard_entry.save()
+
+
+@receiver(post_save, sender=LeaderboardConfiguration)
+def update_leaderboard_scores(sender, instance, **kwargs):
```

Comment:
I don't think we should update leaderboard entries when configuration is changed. It can be very time consuming task and should be done via management command as you have created one below.